### PR TITLE
Exchange based demands

### PR DESCRIPTION
**Pull request by kosecki123**
This PR provides the API for demands management, API for orders cancellation

Features:
- [x] Demand creation
- [x] Order cancellation POST /:id/cancel
- [x] Order GET /:id
- [x] Cancellation/edits, demand states

Known issue where consecutive pause/resume/pause with less that ~1sec demand might end up with orders having wrong states #765

### CODE REVIEW

File: packages/exchange/test/demand.e2e-spec.ts [hunk 1]
```diff
@@ -118,4 +145,115 @@ describe('Demand orders trading', () => {
...
+
+                expect(bid1.price).toBe(createDemandWith2Bids.price);
+                expect(bid2.price).toBe(createDemandWith2Bids.price);
+
+                expect(bid1.demandId).toBe(created.id);
+                expect(bid2.demandId).toBe(created.id);
+            });
+    });
+
+    it('should be able to cancel demand', async () => {
```
Comment (josipbagaric) [inline packages/exchange/test/demand.e2e-spec.ts:170]: Cancelling === Pausing demand?

File: packages/utils-general/src/blockchain-facade/EnumExports.ts [hunk 2]
```diff
@@ -15,3 +15,9 @@ export enum Unit {
     GW = 1e9,
     GWh = 1e9
 }
+
+export enum DemandStatus {
```
Comment (josipbagaric) [inline packages/utils-general/src/blockchain-facade/EnumExports.ts:19]: Can this be moved to the exchange-core package?

**Review by josipbagaric — APPROVED** [review]
LGTM. Only have a few questions, but can otherwise be merged.

File: packages/exchange/test/demand.e2e-spec.ts (hunk 1 above)
Comment (kosecki123) [reply packages/exchange/test/demand.e2e-spec.ts:170]: Yes, typo, should be "pause demand"

File: packages/utils-general/src/blockchain-facade/EnumExports.ts (hunk 2 above)
Comment (kosecki123) [reply packages/utils-general/src/blockchain-facade/EnumExports.ts:19]: I though so too, but exchange-core does not have any notion of the demand, it's bids, asks, orders, trades. Demand is a collection of bids.

**Pull request merged**